sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -416,6 +416,23 @@ object SimplifyConditionals extends Rule[LogicalPlan] with PredicateHelper {
         // these branches can be pruned away
         val (h, t) = branches.span(_._1 != TrueLiteral)
         CaseWhen( h :+ t.head, None)
+
+      case e @ CaseWhen(branches, Some(elseValue)) if {
+        val values = branches.map(_._2) :+ elseValue
+        values.tail.forall(values.head.semanticEquals)
```

Comment:
I think the case is: remove branches that have the same value of `else`:
```
branches.exists(_._2.semanticEquals(elseValue))
```

Reply:
I think what you suggested is the same as the implemented code in the PR, and `elseValue.deterministic` is not required since it is checked in `semanticEquals`.

Being said that, the whole condition can be replaced by `branches.exists(_._2.semanticEquals(elseValue))`

Reply:
I replaced the cond by `branches.forall(_._2.semanticEquals(elseValue))` which is simpler.